cohorts/count.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2016. Mount Sinai School of Medicine
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import numpy as np
+import pandas as pd
+
+from varcode import EffectCollection
+from varcode.effects import Substitution
+
+def snv_count(cohort):
+    sample_variants = cohort.load_variants()
+    def count_func(sample):
+        if sample in sample_variants:
+            return len(sample_variants[sample])
+        return np.nan
+    return count(cohort, count_func, count_col="snv_count")
+
+def nonsynonymous_snv_count(cohort):
+    sample_nonsynonymous_effects = cohort.load_effects(only_nonsynonymous=True)
+    def count_func(sample):
+        if sample in sample_nonsynonymous_effects:
+            return len(sample_nonsynonymous_effects[sample])
+        return np.nan
+    return count(cohort, count_func, count_col="nonsynonymous_snv_count")
+
+def missense_snv_count(cohort):
+    sample_nonsynonymous_effects = cohort.load_effects(only_nonsynonymous=True)
+    sample_missense_effects = dict(
+        [(sample,
+          EffectCollection(
+              [effect for effect in effects if type(effect) == Substitution]))
```

Comment:
Seems like `type` is the thing to use when you don't want inheritance to be factored in: http://stackoverflow.com/questions/1549801/differences-between-isinstance-and-type-in-python